room/compiler/src/main/kotlin/androidx/room/solver/TypeAdapterStore.kt

```diff
@@ -511,6 +512,24 @@ class TypeAdapterStore private constructor(
                         ).process()
                     )
                 }
+
+                // use pojo adapter if we exclude other types.
+                // this happens when @RawQuery or @SkipVerification is used.
+                if (typeMirror.isPojoByExclusion()) {
```

Comment:
It should be enough to only check for `isVoid`, `isVoidObject` and `isKotlinUnit`, everything else, primitives, String, arrays is handled by a `SingleColumnRowAdapter` and one of the built-in type converters. This is done slightly below in this same method ([here](https://github.com/androidx/androidx/blob/androidx-main/room/compiler/src/main/kotlin/androidx/room/solver/TypeAdapterStore.kt#L522)). Which also means we need to keep this logic down there where `runtimeQueryPlaceholder` used to be so that we try to find a single row adapter first before attempting a Pojo one.

Reply:
Thanks for pointing this out, my previous implementation was also missing the checks for `isVoidObject` and `isKotlinUnit`. After reading your comment I agree that it is better to move the Pojo logic back down. 
The original thought was that it might improve readability, but the was hardly the case.